ValveKeyValue/ValveKeyValue/Deserialization/KV1BinaryReader.cs

```diff
@@ -54,79 +54,79 @@ public void Dispose()
             if (!disposed)
             {
                 reader.Dispose();
-                stream.Dispose();
                 disposed = true;
             }
         }
 
         void ReadObjectCore()
         {
-            var type = ReadNextNodeType();
-            var name = ReadNullTerminatedString();
-            ReadValue(name, type);
+            KV1BinaryNodeType type;
+
+            // Keep reading values, until we reach the terminator
+            while ((type = ReadNextNodeType()) != KV1BinaryNodeType.End)
+                ReadValue(type);
```

Comment:
Add braces please (and in other places too)

Reply:
Changed.